x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/Attribute.java

```diff
@@ -137,4 +137,14 @@ public String nodeString() {
     }
 
     protected abstract String label();
+
+    /**
+     * If this field is unsupported this contains the underlying ES types. If there
+     * is a type conflict this will have many elements, some or all of which may
+     * be actually supported types.
+     */
+    @Nullable
+    public List<String> originalTypes() {
```

Comment:
Attribute != Field
This information makes sense only in the case of unsupported fields (potentially union types in the future) and thus makes sense on `FieldAttribute` & co (such as `UnsupportedAttribute`) alone.

Reply:
I'm not sure what you mean.